dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java

```diff
@@ -274,22 +287,34 @@ private void validateTrackedEntityAttributeFilters(
     }
   }
 
-  private void validateGlobalSearchParameters(TrackedEntityQueryParams params, UserDetails user)
-      throws IllegalQueryException {
-    if (!isLocalSearch(params, user)) {
+  private void validatePagination(PageParams pageParams) throws BadRequestException {
+    if (pageParams == null) {
+      return;
+    }
 
+    int systemMaxPageSize = systemSettingsService.getCurrentSettings().getTrackedEntityMaxLimit();
+    if (systemMaxPageSize > 0 && pageParams.getPageSize() > systemMaxPageSize) {
+      throw new BadRequestException(
+          String.format(
+              "Invalid page size: %d. It must not exceed the system limit of KeyTrackedEntityMaxLimit %d.",
+              pageParams.getPageSize(), systemMaxPageSize));
+    }
+  }
+
+  private void validateSearchOutsideCaptureScopeParameters(
+      TrackedEntityQueryParams params, UserDetails user) throws IllegalQueryException {
+    if (!isSearchInCaptureScope(params, user)) {
```

Comment:
Please look for guard clauses like this in general where inverting it and returning early reduces nesting. This is a general thing we can apply to all our code to make it easier to read.

So you can invert it and write it like

```java
if (isSearchInCaptureScope(params, user)) {
  return;
}
```